domain/removal/service/model.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/trace"
+	"github.com/juju/juju/domain/life"
+	modelerrors "github.com/juju/juju/domain/model/errors"
+	"github.com/juju/juju/domain/removal"
+	removalerrors "github.com/juju/juju/domain/removal/errors"
+	"github.com/juju/juju/internal/errors"
+)
+
+// ModelState describes methods for interacting with model state.
+type ModelState interface {
+	// ModelExists returns true if a model exists with the input model
+	// UUID.
+	ModelExists(ctx context.Context, modelUUID string) (bool, error)
+
+	// EnsureModelNotAliveCascade ensures that there is no model identified
+	// by the input model UUID, that is still alive.
+	EnsureModelNotAliveCascade(ctx context.Context, modelUUID string, force bool) (removal.ModelArtifacts, error)
+
+	// ModelScheduleRemoval schedules a removal job for the model with the
+	// input UUID, qualified with the input force boolean.
+	// We don't care if the unit does not exist at this point because:
+	// - it should have been validated prior to calling this method,
+	// - the removal job executor will handle that fact.
+	ModelScheduleRemoval(
+		ctx context.Context, removalUUID, modelUUID string, force bool, when time.Time,
+	) error
+
+	// GetModelLife retrieves the life state of a model.
+	GetModelLife(ctx context.Context, modelUUID string) (life.Life, error)
+
+	// DeleteModelArtifacts deletes all artifacts associated with a model.
+	DeleteModelArtifacts(ctx context.Context, modelUUID string) error
+}
+
+// RemoveModel checks if a model with the input name exists.
+// If it does, the model is guaranteed after this call to be:
+// - No longer alive.
+// - Removed or scheduled to be removed with the input force qualification.
+// The input wait duration is the time that we will give for the normal
+// life-cycle advancement and removal to finish before forcefully removing the
+// model. This duration is ignored if the force argument is false.
+// The UUID for the scheduled removal job is returned.
+func (s *Service) RemoveModel(
+	ctx context.Context,
+	modelUUID model.UUID,
+	force bool,
+	wait time.Duration,
+) (removal.UUID, error) {
+	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	defer span.End()
+
+	exists, err := s.st.ModelExists(ctx, modelUUID.String())
+	if err != nil {
+		return "", errors.Errorf("checking if model exists: %w", err)
+	} else if !exists {
+		return "", errors.Errorf("model does not exist").Add(modelerrors.NotFound)
+	}
+
+	// Ensure the model is not alive.
+	artifacts, err := s.st.EnsureModelNotAliveCascade(ctx, modelUUID.String(), force)
```

Comment:
Unless there is another method `EnsureModelNotAlive`, `Cascade` is superfluous .

Reply:
Cascade indicates, not only does this put the model itself to dying, but it puts everything within that model as dying. This truly does cascade.

The same goes for units, machines and applications.